config/environments/test.rb

```diff
@@ -22,7 +22,7 @@
     env.cache = ActiveSupport::Cache.lookup_store(:memory_store)
   end
 
-  config.assets.compile = ENV['TEST_SUITE'] == 'spec:javascript'
```

Comment:
Note, if we keep this logic, it will have to also add the spec and api test suites since they would also need to precompile their assets instead of live compile them.

It feels like live compilation is less intrusive.